tests/factoranalysis/test_htfa.py

```diff
@@ -126,37 +126,39 @@ def test_can_run():
         if idx % size == rank:
             my_R.append(R[idx])
 
-    htfa = HTFA(
-        K,
-        n_subj=n_subj,
-        max_global_iter=max_global_iter,
-        max_local_iter=max_local_iter,
-        max_voxel=max_voxel,
-        max_tr=max_tr,
-        verbose=True)
-    assert htfa, "Invalid HTFA instance!"
-
-    X = []
-    for s in np.arange(n_subj):
-        X.append(np.random.rand(n_voxel, n_tr))
-    my_data = []
-    for idx in np.arange(n_subj):
-        if idx % size == rank:
-            my_data.append(X[idx])
-
-    if rank == 0:
-        htfa.fit(my_data, R=my_R)
-        assert True, "Root successfully running HTFA"
-        assert htfa.global_prior_.shape[0] == htfa.prior_bcast_size,\
-            "Invalid result of HTFA! (wrong # element in global_prior)"
-        assert htfa.global_posterior_.shape[0] == htfa.prior_bcast_size,\
-            "Invalid result of HTFA! (wrong # element in global_posterior)"
-
-    else:
-        htfa.fit(my_data, R=my_R)
-        assert True, "worker successfully running HTFA"
-        print(htfa.local_weights_.shape)
-        assert htfa.local_weights_.shape[0] == n_tr * K,\
-            "Invalid result of HTFA! (wrong # element in local_weights)"
-        assert htfa.local_posterior_.shape[0] == htfa.prior_size,\
-            "Invalid result of HTFA! (wrong # element in local_posterior)"
+    for root in range(size):
```

Comment:
You are right. The MPICH `mpiexec` manual says:
`mpiexec returns the maximum of the exit status values of all of the processes created by mpiexec.`
http://www.mpich.org/static/docs/latest/www1/mpiexec.html

However, we can use the `-print-all-exitcodes` option to get all values:
```
mihai@alb:~$ mpiexec -n 3 -print-all-exitcodes hostname
alb
alb
alb
[mpiexec@alb] Exit codes: [alb] 0,0,0
```
If one of the processes returns 0, the test passes. This way, we can always run the minimum amount of tests. In this case, we only have 2 code paths, 1 for the master, 1 for the workers, so 2 passes are enough, irrespective of the number of processes.

BTW, I referenced `mpiexec`, not `mpirun`, because this is what the MPI standard specifies:
https://www.mpi-forum.org/docs/mpi-3.1/mpi31-report/node228.htm
Yida is working on getting Intel MPI 2017 working on Metacortex so we can switch to `mpiexec`.